mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,265 @@
+        ...         for i, gt_label in enumerate(gt_labels):
+        ...             begin = i * 300
+        ...             end = begin + 300
+        ...             pred_instance_mask[begin:end] = i
+        ...             labels.append(gt_label)
+        ...             scores.append(.99)
+        ...
+        ...         results_dict['pts_instance_mask'] = torch.tensor(
+        ...             pred_instance_mask)
+        ...         results_dict['instance_labels'] = torch.tensor(labels)
+        ...         results_dict['instance_scores'] = torch.tensor(scores)
+        ...
+        ...         predictions.append(results_dict)
+        ...         groundtruths.append(ann_info_data)
+        ...
+        ...     return predictions, groundtruths
+        >>>
+        >>> instance_seg_metric = InstanceSegMetric(dataset_meta=dataset_meta)
+        >>> res = instance_seg_metric(predictions, groundtruths)
+        >>> res
+        {
+        'all_ap': 1.0,
+        'all_ap_50%': 1.0,
+        'all_ap_25%': 1.0,
+        'classes': {
+            'cabinet': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            'bed': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            'chair': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            ...
+        }
+    }
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        assert self.dataset_meta is not None
+        self.classes = self.dataset_meta['classes']
+        self.valid_class_ids = self.dataset_meta['seg_valid_class_ids']
+
+    def add(self, predictions: Sequence[Dict], groundtruths: Sequence[Dict]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Process one batch of data samples and predictions.
+
+        The processed results should be stored in ``self.results``,
+        which will be used to compute the metrics when all batches
+        have been processed.
+        Args:
+            predictions (Sequence[Dict]): A sequence of dict. Each dict
+                representing a detection result, with the following keys:
+
+            - pts_instance_mask(numpy.ndarray): Predicted instance masks.
+            - instance_labels(numpy.ndarray): Predicted instance labels.
+            - instance_scores(numpy.ndarray): Predicted instance scores.
+            groundtruths (Sequence[Dict]): A sequence of dict. Each dict
+                represents a groundtruths for an image, with the following
+                keys:
+            - pts_instance_mask(numpy.ndarray): Ground truth instance masks.
+            - pts_semantic_mask(numpy.ndarray): Ground truth semantic masks.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
+            self._results.append((deepcopy(prediction), deepcopy(groundtruth)))
```

Comment:
Why we need deepcopy~

Reply:
If we don't use deepcopy, it will change the value of it which may cause confusion.